cql3/attributes.cc

```diff
@@ -38,17 +40,14 @@ bool attributes::is_timeout_set() const {
 }
 
 int64_t attributes::get_timestamp(int64_t now, const query_options& options) {
-    if (!_timestamp.has_value()) {
+    if (!_timestamp.has_value() || _timestamp_unset_guard.should_skip_operation(options)) {
```

Comment:
Continuing my question above about the misleading method name "should_skip_operation", I think this code will be clearer with instead of using a "guard" object, you could use just a function call which makes it clearer it is called before evaluate just just to check unset, e.g.,
`if (!_timestamp.has_value || expr_is_unset(options)) return now`.
The expr_is_unset() function will return true only if the expression is UNSET_VALUE and nothing else - if it's something more complicated, expr::evalute() will throw below (if I understand correctly).

Reply:
The guard does two things:

1. during prepare time, it extracts the expression to check and casts it into a bind_variable.
2. at execution time, it uses the prepared information to quickly check.

So it needs to remain an object, just a better named one.

Note an expression can't be unset. An expression can be a bind variable, and the bind variable can be bound to UNSET.

Reply:
Right - by "expression **is** unset" I meant that the expression is entirely one unset bound variable (and not some larger expression which may or may not have unset bound variables). I don't know how to name it.